src/models/db.js

```diff
@@ -28,6 +28,17 @@ const IDLE           = 0
 const INITIALIZING   = 1;
 const READY          = 2;
 
+const hooks = [
+  'beforeValidate',
+  'afterValidate',
+  'beforeCreate',
+  'afterCreate',
+  'beforeDestroy',
+  'afterDestroy',
+  'beforeUpdate',
+  'afterUpdate'
+];
```

Comment:
I removed `beforeRestore`, and `afterRestore`. My thinking was they would not be run during the use of our ST API and therefore it would be dead code to add the hooks and let the clients set them.